src/test/kotlin/tech/relaycorp/relaynet/ramf/RAMFMessageTest.kt

```diff
@@ -0,0 +1,128 @@
+                assertEquals(5, sequence.size())
+            }
+
+            @Test
+            fun `Recipient should be stored as an ASN1 VisibleString`() {
+                val sequence = getAsn1Sequence()
+                val recipientRaw = sequence.getObjectAt(0) as DLTaggedObject
+                val recipientDer = DERVisibleString.getInstance(recipientRaw, false)
+                assertEquals(stubRamfMessage.recipient, recipientDer.string)
+            }
+
+            @Test
+            fun `Message id should be stored as an ASN1 VisibleString`() {
+                val sequence = getAsn1Sequence()
+                val messageIdRaw = sequence.getObjectAt(1) as DLTaggedObject
+                val messageIdDer = DERVisibleString.getInstance(messageIdRaw, false)
+                assertEquals(stubRamfMessage.messageId, messageIdDer.string)
+            }
+
+            @Test
+            fun `Creation time should be stored as an ASN1 DateTime`() {
+                val sequence = getAsn1Sequence()
+                val creationTimeRaw = sequence.getObjectAt(2) as DLTaggedObject
+                // We should technically be using a DateTime type instead of GeneralizedTime, but BC
+                // doesn't support it.
+                val creationTimeDer = DERGeneralizedTime.getInstance(creationTimeRaw, false)
+                val dateTimeFormatter = DateTimeFormatter.ofPattern("yyyyMMddHHmmss")
+                assertEquals(
+                        stubRamfMessage.creationTimeUtc.format(dateTimeFormatter),
+                        creationTimeDer.timeString
+                )
+            }
+
+            @Test
+            fun `TTL should be stored as an ASN1 Integer`() {
+                val sequence = getAsn1Sequence()
+                val ttlRaw = sequence.getObjectAt(3) as DLTaggedObject
+                val ttlDer = ASN1Integer.getInstance(ttlRaw, false)
+                assertEquals(stubRamfMessage.ttl, ttlDer.intPositiveValueExact())
+            }
+
+            @Test
+            fun `Payload should be stored as an ASN1 Octet String`() {
+                val sequence = getAsn1Sequence()
+                val payloadRaw = sequence.getObjectAt(4) as DLTaggedObject
+                val payloadDer = ASN1OctetString.getInstance(payloadRaw, false)
+                assertEquals(stubRamfMessage.payload.asList(), payloadDer.octets.asList())
+            }
+        }
+    }
+
+    @Nested
+    inner class Deserialize {
+        @Nested
+        inner class CreationTime {
+            @Test
+            @Disabled("Pending implementation")
+            fun `A timezone other than UTC should not be allowed`() {
+            }
+
+            @Test
+            @Disabled("Pending implementation")
+            fun `Timezone may be unset`() {
+            }
+
+            @Test
+            @Disabled("Pending implementation")
+            fun `Timezone may be set to UTC`() {
+            }
+        }
+    }
+}
```

Comment:
This is technically part of #9 but I thought of those edge cases and thought it'd be good to create placeholders for them. I probably should've added a comment in #9 in hindsight.

Reply:
the TTL you have is for 180 days rather than 90